File: RedSheet.DbEntities/Models/vProjectBackGroundRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectBackGroundRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectBackGroundRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string Focus { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int NegotiationTypeId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OpponentName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectBackgroundId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Reason { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ValueObjectiveId { get; set; } 

	public vProjectBackGroundRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-back-ground-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectBackGroundRecord {
    constructor(vProjectBackGroundRecord?: vProjectBackGroundRecord )  {
        let properties = [ "focus", "negotiationTypeId", "opponentName", "projectBackgroundId", "projectModuleId", "reason", "valueObjectiveId",];
        for (let property of properties)
            if (vProjectBackGroundRecord && vProjectBackGroundRecord[property])
                this[property] = vProjectBackGroundRecord[property];
    }
 
    @required()
    @maxLength(1000)
	focus : string =   undefined;
 
	negotiationTypeId : number =   0 ;
 
    @required()
    @maxLength(1000)
	opponentName : string =   undefined;
 
    @range(1,2147483647)
	projectBackgroundId : number =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(1000)
	reason : string =   undefined;
 
    @range(1,2147483647)
	valueObjectiveId : number =   undefined;


}
